commit 58b069dfc6198931ed7eabc82a40c220f88729dc (refs/changes/51/3851/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-04-20T21:44:30+00:00 (1 year, 6 months ago)
    
    joyent/node-smartdc-auth#16 KeyPair isKey assertion should be min-versioned
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/keypair.js b/lib/keypair.js
index bf644ed..4ff597e 100644
--- a/lib/keypair.js
+++ b/lib/keypair.js
@@ -47,7 +47,11 @@ function SDCKeyPair(kr, opts) {
 	this.comment = '';
 
 	if (opts.public !== undefined) {
-		mod_assert.ok(mod_sshpk.Key.isKey(opts.public),
+		/*
+		 * We need a Key of v1,3 or later for defaultHashAlgorithm and
+		 * the Signature hashAlgorithm support.
+		 */
+		mod_assert.ok(mod_sshpk.Key.isKey(opts.public, [1, 3]),
 		    'options.public must be a sshpk.Key instance');
 		this.comment = opts.public.comment;
 	}
